shared/lib_battery_lifetime_calendar_cycle.h

```diff
@@ -70,6 +70,13 @@ struct cycle_state {
     int rainflow_jlt;                       // last index in Peaks, i.e, if Peaks = [0,1], then jlt = 1
     std::vector<double> rainflow_peaks;
 
+    // values for current day, refreshed end of each day; used only in analytic life models not calendar_cycle model
```

Comment:
This header file could also use some additional comments on constructor usage. For example, is it correct that initializing with only params is only used for first time setup, where as params and state are for resuming from a stateful run?